src/Operation/CreateSearchIndexes.php

```diff
@@ -0,0 +1,104 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB\Operation;
+
+use MongoDB\Driver\Command;
+use MongoDB\Driver\Exception\RuntimeException as DriverRuntimeException;
+use MongoDB\Driver\Server;
+use MongoDB\Exception\InvalidArgumentException;
+use MongoDB\Exception\UnsupportedException;
+use MongoDB\Model\SearchIndexInput;
+
+use function array_is_list;
+use function array_map;
+use function current;
+use function is_array;
+use function is_object;
+use function sprintf;
+
+/**
+ * Operation for the createIndexes command.
+ *
+ * @see \MongoDB\Collection::createSearchIndex()
+ * @see \MongoDB\Collection::createSearchIndexes()
+ * @see https://mongodb.com/docs/manual/reference/command/createSearchIndexes/
+ */
+class CreateSearchIndexes implements Executable
+{
+    /** @var string */
+    private $databaseName;
+
+    /** @var string */
+    private $collectionName;
+
+    /** @var array */
+    private $indexes = [];
+
+    /**
+     * Constructs a createSearchIndexes command.
+     *
+     * @param string             $databaseName   Database name
+     * @param string             $collectionName Collection name
+     * @param list<array|object> $indexes        List of search index specifications
+     * @throws InvalidArgumentException for parameter parsing errors
+     */
+    public function __construct(string $databaseName, string $collectionName, array $indexes)
+    {
+        if (! array_is_list($indexes)) {
+            throw new InvalidArgumentException('$indexes is not a list');
+        }
+
+        foreach ($indexes as $i => $index) {
+            if (is_object($index)) {
+                $index = (array) $index;
+            } elseif (! is_array($index)) {
+                throw InvalidArgumentException::invalidType(sprintf('$index[%d]', $i), $index, 'array or object');
+            }
+
+            $this->indexes[] = new SearchIndexInput($index);
+        }
+
+        $this->databaseName = $databaseName;
+        $this->collectionName = $collectionName;
+    }
+
+    /**
+     * Execute the operation.
+     *
+     * @see Executable::execute()
+     * @return string[] The names of the created indexes
+     * @throws UnsupportedException if write concern is used and unsupported
+     * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
+     */
+    public function execute(Server $server): array
+    {
+        $cmd = [
+            'createSearchIndexes' => $this->collectionName,
+            'indexes' => $this->indexes,
```

Comment:
Note: CreateIndexes supports a `comment` option, which it would inject here.

Reply:
I just came across this previously resolved comment after leaving suggestions to unwrap the loops for `comment` options in other files. Feel free to do so here as well, or just keep everything as-is and decline the suggestions accordingly.

Also, you can resolve this thread again after reading.